### ISSUE TITLE
Crasher

### ISSUE THREAD

**Issue by zenspider** [ahoward/tagz#5]
I was finally able to get a minimal repro.

The following code will cause a crash:

```ruby
require "tagz"
include Tagz

require "anything"
```

I tracked it down to this part of `Tagz.method_missing`:

```ruby
        if(strategy.nil? or (tagz.nil? and Tagz.privately===self))
          begin
            return super
          ensure
            $!.set_backtrace caller(1) if $!
          end
        end
```

If you change from `caller(1)` to `caller[1..-1]` it works fine. The rdoc seems to agree with your code tho, and irb seems happy with it too. So I'm not sure what's going on.

This IS still different from the MASSIVE C-level crash I was getting yesterday, but it is a start.

**Comment by ahoward** [comment]
wow - ok.  i'll look soon - thanks!

**Comment by ahoward** [comment]
@zenspider basically i'm thinking that my idea about 'cleaning up the backtrace' is just fucking lame/stupid/error-prone and doing *nothing* here is better code.  agree?

**Comment by zenspider** [comment]
For something like tagz? Yeah, probably. That said, I'd love to know WHY `caller[1]` is fine but `caller(1)` causes a blowup. But that can be a separate discovery.

**Comment by ahoward** [comment]
fixed in 9.10.0.   gem pushed.
